exporter/opensearchexporter/internal/objmodel/objmodel_test.go

```diff
@@ -0,0 +1,435 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package objmodel
+
+import (
+	"math"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/elastic/go-structform/json"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+var dijkstra = time.Date(1930, 5, 11, 16, 33, 11, 123456789, time.UTC)
+
+func TestObjectModel_CreateMap(t *testing.T) {
+	tests := map[string]struct {
+		build func() Document
+		want  Document
+	}{
+		"from empty map": {
+			build: func() Document {
+				return DocumentFromAttributes(pcommon.NewMap())
+			},
+		},
+		"from map": {
+			build: func() Document {
+				m := pcommon.NewMap()
+				m.PutInt("i", 42)
+				m.PutStr("str", "test")
+				return DocumentFromAttributes(m)
+			},
+			want: Document{[]field{{"i", IntValue(42)}, {"str", StringValue("test")}}},
+		},
+		"ignores nil values": {
+			build: func() Document {
+				m := pcommon.NewMap()
+				m.PutEmpty("null")
+				m.PutStr("str", "test")
+				return DocumentFromAttributes(m)
+			},
+			want: Document{[]field{{"str", StringValue("test")}}},
+		},
+		"from map with prefix": {
+			build: func() Document {
+				m := pcommon.NewMap()
+				m.PutInt("i", 42)
+				m.PutStr("str", "test")
+				return DocumentFromAttributesWithPath("prefix", m)
+			},
+			want: Document{[]field{{"prefix.i", IntValue(42)}, {"prefix.str", StringValue("test")}}},
+		},
+		"add attributes with key": {
+			build: func() (doc Document) {
+				m := pcommon.NewMap()
+				m.PutInt("i", 42)
+				m.PutStr("str", "test")
+				doc.AddAttributes("prefix", m)
+				return doc
+			},
+			want: Document{[]field{{"prefix.i", IntValue(42)}, {"prefix.str", StringValue("test")}}},
+		},
+		"add attribute flattens a map value": {
+			build: func() (doc Document) {
+				mapVal := pcommon.NewValueMap()
+				m := mapVal.Map()
+				m.PutInt("i", 42)
+				m.PutStr("str", "test")
+				doc.AddAttribute("prefix", mapVal)
+				return doc
+			},
+			want: Document{[]field{{"prefix.i", IntValue(42)}, {"prefix.str", StringValue("test")}}},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			doc := test.build()
+			doc.Sort()
+			assert.Equal(t, test.want, doc)
+		})
+	}
+}
+
+func TestDocument_Sort(t *testing.T) {
+	tests := map[string]struct {
+		build func() Document
+		want  Document
+	}{
+		"keys are sorted": {
+			build: func() (doc Document) {
+				doc.AddInt("z", 26)
+				doc.AddInt("a", 1)
+				return doc
+			},
+			want: Document{[]field{{"a", IntValue(1)}, {"z", IntValue(26)}}},
+		},
+		"sorting is stable": {
```

Comment:
Is this a required quality, as long as the sorting provides the same result given the same input?